src/bindings/swift/gen_swift.rs

```diff
@@ -0,0 +1,498 @@
+    }
+}
+
+/// Filters for our Askama templates above. These output C (for the bridging
+/// header) and Swift (for the actual library) declarations.
+mod filters {
+    use super::*;
+    use std::fmt;
+
+    /// Declares a C type in the bridging header.
+    pub fn decl_c(type_: &TypeReference) -> Result<String, askama::Error> {
+        Ok(match type_ {
+            // These native types map nicely to the FFI without conversion.
+            TypeReference::U32 => "uint32_t".into(),
+            TypeReference::U64 => "uint64_t".into(),
+            TypeReference::Float => "float".into(),
+            TypeReference::Double => "double".into(),
+            TypeReference::Bytes => "RustBuffer".into(),
+            // Our FFI lowers Booleans into bytes, to work around JNA bugs.
+            // We'll lift these up into Booleans on the Swift side.
+            TypeReference::Boolean => "uint8_t".into(),
+            // These types need conversion, and special handling for lifting/lowering.
+            TypeReference::Enum(_) => "uint32_t".into(),
+            TypeReference::Record(_) => "RustBuffer".into(),
+            TypeReference::Optional(_) => "RustBuffer".into(),
+            TypeReference::Object(_) => "uint64_t".into(),
+            _ => panic!("[TODO: decl_c({:?})", type_),
+        })
+    }
+
+    /// Declares a Swift type in the public interface for the library.
+    pub fn decl_swift(type_: &TypeReference) -> Result<String, askama::Error> {
+        Ok(match type_ {
+            TypeReference::U32 => "UInt32".into(),
+            TypeReference::U64 => "UInt64".into(),
+            TypeReference::Float => "Float".into(),
+            TypeReference::Double => "Double".into(),
+            // TypeReference::Bytes => "Data".into(),
+            TypeReference::Boolean => "Bool".into(),
+            TypeReference::Enum(name) => name.into(),
+            TypeReference::Record(name) => name.into(),
+            TypeReference::Optional(type_) => format!("{}?", decl_swift(type_)?),
+            TypeReference::Object(name) => name.into(),
+            _ => panic!("[TODO: decl_swift({:?})", type_),
+        })
+    }
+
+    /// Lowers a Swift type into a C type. This is used to pass arguments over
+    /// the FFI, from Swift to Rust.
+    pub fn lower_swift(
+        name: &dyn fmt::Display,
+        _type_: &TypeReference,
+    ) -> Result<String, askama::Error> {
+        Ok(format!("{}.lower()", name))
+    }
+
+    /// ...
+    pub fn lift_from_swift(
+        name: &dyn fmt::Display,
+        type_: &TypeReference,
+    ) -> Result<String, askama::Error> {
+        Ok(format!("{}.lift(from: {})", decl_swift(type_)?, name))
+    }
+
+    /// ...
+    pub fn lift_swift(
+        name: &dyn fmt::Display,
+        type_: &TypeReference,
+    ) -> Result<String, askama::Error> {
+        Ok(format!("{}.lift({})", decl_swift(type_)?, name))
+    }
+
+    /// ...
+    pub fn decl_enum_variant_swift(name: &str) -> Result<String, askama::Error> {
```

Comment:
I like what @jhugman is doing with naming conventions over in #2! This was a pretty specific hack to turn `SCREAMING_CASE` enum variant names into `camelCase`; it would be amazing to have something more generic!